File: sdk/typescript/tools/restart-acceptance.mjs
import assert from 'node:assert/strict';
import process from 'node:process';
const packageName = '@capsem/sdk';
/** @type {unknown} */
const built = await import(packageName);
const {Hypervisor, RestartAuthentication, RestartStatus, ServiceManager} =
  /** @type {typeof import('../src/index.js')} */ (built);

const url = process.env.SDK_GATEWAY_URL, token = process.env.SDK_GATEWAY_TOKEN;
assert(url && token, 'SDK restart fixture must provide gateway credentials');
const hv = new Hypervisor(url, token);
try {
  const response = await hv.restart();
  assert.equal(response.status, RestartStatus.ACCEPTED);
  assert.equal(response.authentication, RestartAuthentication.NEW_TOKEN_REQUIRED);
  assert.equal(response.manager, ServiceManager.LAUNCHD);
} finally {
  hv.close();
}
process.stdout.write('SDK_RESTART_ACCEPTED\n');

File: sdk/python/tests/restart_acceptance.py
"""A managed restart is sent once through the public Python facade."""

import asyncio
import os

from capsem import Hypervisor, models


async def main() -> None:
    async with Hypervisor(os.environ["SDK_GATEWAY_URL"], os.environ["SDK_GATEWAY_TOKEN"]) as hv:
        response = await hv.restart()
        assert response.status is models.RestartStatus.ACCEPTED
        assert response.authentication is models.RestartAuthentication.NEW_TOKEN_REQUIRED
        assert response.manager is models.ServiceManager.LAUNCHD
    print("SDK_RESTART_ACCEPTED")


if __name__ == "__main__":
    asyncio.run(main())
